lib/web/resources.go

```diff
@@ -329,14 +348,187 @@ func listResources(clt resourcesAPIGetter, r *http.Request, resourceKind string)
 	return clt.ListResources(r.Context(), req)
 }
 
+func handleClusterNodesGet(clt resourcesAPIGetter, r *http.Request, clusterName string, userRoles services.RoleSet) (*listResourcesGetResponse, error) {
+	resp, err := listResources(clt, r, types.KindNode)
+	if err == nil {
+		servers, err := types.ResourcesWithLabels(resp.Resources).AsServers()
+		if err != nil {
+			return nil, trace.Wrap(err)
+		}
+
+		return &listResourcesGetResponse{
+			Items:      ui.MakeServers(clusterName, servers, userRoles),
+			StartKey:   &resp.NextKey,
+			TotalCount: &resp.TotalCount,
+		}, nil
+	}
+
+	if trace.IsNotImplemented(err) {
+		nodes, err := clt.GetNodes(r.Context(), apidefaults.Namespace)
+		if err != nil {
+			return nil, trace.Wrap(err)
+		}
+
+		return &listResourcesGetResponse{
+			Items: ui.MakeServers(clusterName, nodes, userRoles),
+		}, nil
+	}
+	return nil, trace.Wrap(err)
```

Comment:
Suggestion: cut nesting a bit and avoid carrying `err` all the way to the end (if it's not NotImplemented).

```suggestion
	if !trace.IsNotImplemented(err) {
		return nil, trace.Wrap(err)
	}

	// Fallback to GetNodes.
	nodes, err := clt.GetNodes(r.Context(), apidefaults.Namespace)
	if err != nil {
		return nil, trace.Wrap(err)
	}
	return &listResourcesGetResponse{
		Items: ui.MakeServers(clusterName, nodes, userRoles),
	}, nil
```